x-pack/plugins/security/server/authorization/check_privileges.ts

```diff
@@ -111,23 +157,29 @@ export function checkPrivilegesWithRequestFactory(
       return {
         hasAllRequested: hasPrivilegesResponse.has_all_requested,
         username: hasPrivilegesResponse.username,
-        privileges: privilegeArray,
+        privileges: {
+          kibana: privilegeArray,
+          elasticsearch: {
+            cluster: clusterPrivileges,
+            index: indexPrivileges,
+          },
+        },
       };
     };
 
     return {
-      async atSpace(spaceId: string, privilegeOrPrivileges: string | string[]) {
+      async atSpace(spaceId: string, privileges: CheckPrivilegesPayload) {
```

Comment:
**note**: I don't love what I have here. Asking to check privileges at a space (or spaces) only makes sense within the context of Kibana privileges, but we now allow you to check Elasticsearch privileges as part of this call.

The space(s) you choose to check within have no bearing whatsoever on the result of the cluster/index privilege check.

I think it would be worthwhile to revisit this interface, but I didn't want to take that on as part of this PR given its current size. A followup task to refactor this would probably be worthwhile. What do you think?